source/numerics/data_out_dof_data.cc

```diff
@@ -986,13 +987,13 @@ add_data_vector (const VectorType                         &vec,
       Assert (false, ExcInternalError());
     }
 
-  internal::DataOut::DataEntryBase<DoFHandlerType> *new_entry
-    = new internal::DataOut::DataEntry<DoFHandlerType,VectorType>(dofs, &vec, names,
-        data_component_interpretation);
+  auto new_entry = std_cxx14::make_unique<internal::DataOut::DataEntry<DoFHandlerType,VectorType>>
+                   (dofs, &vec, names, data_component_interpretation);
+
   if (actual_type == type_dof_data)
-    dof_data.push_back (std::shared_ptr<internal::DataOut::DataEntryBase<DoFHandlerType> >(new_entry));
+    dof_data.emplace_back (std::move(new_entry));
```

Comment:
I'm surprised that for `std::unique_ptr` objects it makes any kind of difference whether (i) you use `emplace_back` over `push_back`, and (ii) whether you do or do not move -- any kind of copy operation for `std::unique_ptr` object is a move operation.